xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
Nice, and the actor stuff is already in util. It's also very optimized, I see. But I'm not sure it would solve the problem I have here and why I made the custom signal stuff in the first place:

Windowing is asked for a new video sync implementation, but it does not own it afterwards - the caller does. And from the interface it is not guaranteed that only one such video sync objects exists - so theoretically, there could be two or more video sync objects active at the same time. They all need to get the presented event, so the signal mechanism must support multiple subscribers.

Otherwise, I'd have just used a plain interface or single std::function.